rules/library.bzl

```diff
@@ -767,23 +776,33 @@ def apple_library(
         )
         import_vfsoverlays.append(import_name + "_vfs")
 
+    vendored_deps_contains_arm64_sim_slice = {}
     for xcframework in kwargs.pop("vendored_xcframeworks", []):
         xcframework_name, xcframework_name_vfs = _xcframework(library_name = name, **xcframework)
+        vendored_deps_contains_arm64_sim_slice[xcframework_name] = _xcframework_contains_arm64_sim_slice(xcframework)
         vendored_deps.append(xcframework_name)
         import_vfsoverlays.append(xcframework_name_vfs)
 
     # We don't currently support dynamic library
     for vendored_dynamic_library in kwargs.pop("vendored_dynamic_libraries", []):
         fail("no support for dynamic library: %s" % vendored_dynamic_library)
 
-    # TODO(jmarino)Perhaps it uses a import_middleman here
-    if len(vendored_deps):
-        import_middleman(name = name + ".import_middleman", deps = vendored_deps, tags = ["manual"])
+    vendored_deps_non_arm64_sim = [d for d in vendored_deps if not vendored_deps_contains_arm64_sim_slice.get(d, False)]
```

Comment:
It was able to handle both xcframeworks and non deps at a time - let me see if we can preserve this